index.js

```diff
@@ -1,89 +1,106 @@
-'use strict';
-var mqtt = require('mqtt');
-var inArray = require('in-array');
-
-var RegularClientPrototype = mqtt.MqttClient.prototype;
-
-var ASYNC_METHODS = ['publish',
-  'subscribe',
-  'unsubscribe',
-  'end'
-];
-
-var SYNC_METHODS = [
-  'addListener',
-  'emit',
-  'eventNames',
-  'getMaxListeners',
-  'listenerCount',
-  'listeners',
-  'off',
-  'on',
-  'once',
-  'prependListener',
-  'prependOnceListener',
-  'removeAllListeners',
-  'removeListener',
-  'setMaxListeners',
-  'rawListeners'
-];
+'use strict'
 
-module.exports = {
-  connect: connect,
-  AsyncClient: AsyncClient
-};
+const mqtt = require('mqtt');
 
-function connect (brokerURL, opts) {
-  var client = mqtt.connect(brokerURL, opts);
+const RegularClientPrototype = mqtt.MqttClient.prototype;
 
-  var asyncClient = new AsyncClient(client);
-
-  return asyncClient;
-}
-
-function AsyncClient (client) {
-  this._client = client;
-}
+class AsyncClient {
+  constructor (client) {
+    this._client = client;
+  }
 
-AsyncClient.prototype = {
   set handleMessage (newHandler) {
     this._client.handleMessage = newHandler;
-  },
+  }
+
   get handleMessage () {
     return this._client.handleMessage;
   }
-};
 
-ASYNC_METHODS.forEach(defineAsync);
-SYNC_METHODS.forEach(definePassthrough);
+  async publish (...args) {
+    return this._client.publish(...args);
```

Comment:
This still isn't passing a callback to the client. It should look something like this:

```suggestion
    return new Promise((resolve, reject) => {
      this._client.publish(...args, (err, result) => {
        if(err) reject(err)
        else resolve(result)
      })
    })
```

This will return a promise that only resolves once the publish has been acknowledged by the broker, or will reject if there was a problem.

Without this wrapper, it'd be resolving immediately and the potential error would never be handled.

This should be done with all the async methods.